File: frontend/src/types/diary.ts
/**
 * 日記作成リクエスト
 */
export interface DiaryCreateRequest {
  /** 日記のタイトル (1-200文字) */
  title: string;
  /** 日記の内容 (1文字以上) */
  content: string;
}

/**
 * 日記レスポンス
 */
export interface DiaryResponse {
  /** 日記ID */
  id: number;
  /** 日記のタイトル */
  title: string;
  /** 日記の内容 */
  content: string;
  /** 作成日時 */
  created_at: string;
  /** 更新日時 */
  updated_at: string;
}

/**
 * 日記一覧レスポンス
 */
export interface DiariesListResponse {
  /** 日記一覧 */
  diaries: DiaryResponse[];
  /** 総件数 */
  total_count: number;
}

File: backend/app/src/schemas/diary.py
from datetime import datetime
from typing import List

from pydantic import BaseModel, Field


class DiaryCreateRequest(BaseModel):
    """日記作成リクエスト"""

    title: str = Field(..., min_length=1, max_length=200, description="日記のタイトル")
    content: str = Field(..., min_length=1, description="日記の内容")


class DiaryResponse(BaseModel):
    """日記レスポンス"""

    id: int = Field(..., description="日記ID")
    title: str = Field(..., description="日記のタイトル")
    content: str = Field(..., description="日記の内容")
    created_at: datetime = Field(..., description="作成日時")
    updated_at: datetime = Field(..., description="更新日時")

    class Config:
        from_attributes = True


class DiariesListResponse(BaseModel):
    """日記一覧レスポンス"""

    diaries: List[DiaryResponse] = Field(..., description="日記一覧")
    total_count: int = Field(..., description="総件数")

    class Config:
        from_attributes = True
